CMakeLists.txt

```diff
@@ -49,6 +49,28 @@ function(packageProject)
     endif()
 
     if(DEFINED PROJECT_VERSION_HEADER)
+      string(REPLACE "." ";" VERSION_LIST "${PROJECT_VERSION}")
+      
+      list(POP_FRONT VERSION_LIST 
+        PROJECT_VERSION_MAJOR
+        PROJECT_VERSION_MINOR
+        PROJECT_VERSION_PATCH
+        PROJECT_VERSION_TWEAK
+      )
+
+      if(NOT DEFINED PROJECT_VERSION_MAJOR)
+        set(PROJECT_VERSION_MAJOR "0")
+      endif()
+      if(NOT DEFINED PROJECT_VERSION_MINOR)
+        set(PROJECT_VERSION_MINOR "0")
+      endif()
+      if(NOT DEFINED PROJECT_VERSION_PATCH)
+        set(PROJECT_VERSION_PATCH "0")
+      endif()
+      if(NOT DEFINED PROJECT_VERSION_TWEAK)
+        set(PROJECT_VERSION_TWEAK "0")
+      endif()
```

Comment:
I wonder if zero is a good default, or rather an empty string? What does CMake do?

Reply:
CMake creates empty strings, but I think zero is a good because it gives a concrete value to it. It would also break, e.g.:
```cpp
std::cout << PROJECT_VERSION_PATCH << '\n';
```
or any case where you need a value to actually exist.

Reply:
Bumping the conversation up

Reply:
Oh hey sorry, I forgot to respond! Yeah I just noticed that the versions are being defined as integers and not string in the C header, which probably makes sense for version comparisons. In that case I think 0 is a reasonable default.